src/tracer.rs

```diff
@@ -26,15 +26,11 @@ pub struct Tracer<S, T> {
 }
 impl<S: Sampler<T>, T> Tracer<S, T> {
     /// Makes a new `Tracer` instance.
-    pub fn new(sampler: S) -> (Self, SpanReceiver<T>) {
-        let (tx, rx) = mpsc::channel();
-        (
-            Tracer {
-                sampler: Arc::new(sampler),
-                span_tx: tx,
-            },
-            rx,
-        )
+    pub fn new(sampler: S, span_tx: SpanSender<T>) -> Self {
```

Comment:
For keeping compatibility with the existing code, I think it is better to provide this function as another constructor as follows:
```rust
// This constructor has the same interface with the previous versions (except the definition of `SpanReceiver`).
pub fn new(sampler: S) -> (Self, SpanReceiver<T>) {
    let (span_tx, span_rx) = crossbeam_channel::unbounded();
    (Self::with_sender(sampler, span_tx), span_rx)
}

// New interface that is more flexible.
pub fn with_sender(sampler: S, span_tx: SpanSender<T>) -> Self {
    Tracer {
        sampler: Arc::new(sampler),
        span_tx,
     }
}
```

Reply:
Constructor added.